### PR TITLE
Ruby 3 support

### PR DESCRIPTION
**Pull request by atog**
I ran into a problem (wrong number of arguments) passing optional params to the variables endpoint using Ruby 3.

The problem related to this change: Keyword arguments are separated from other arguments (https://www.ruby-lang.org/en/news/2020/12/25/ruby-3-0-0-released/).

### CODE REVIEW

File: lib/gitlab.rb [hunk 1]
```diff
@@ -21,11 +21,16 @@ def self.client(options = {})
...
   end
 
-  # Delegate to Gitlab::Client
-  def self.method_missing(method, *args, &block)
-    return super unless client.respond_to?(method)
-
-    client.send(method, *args, &block)
+  if RUBY_VERSION >= '3.0.0'
+    def self.method_missing(method, *args, **keywargs, &block)
+      return super unless client.respond_to?(method)
```
Comment (NARKOZ) [inline lib/gitlab.rb:26]: Could you add an empty line after just to fix the rubocop offense?

File: lib/gitlab.rb [hunk 2]
```diff
@@ -21,11 +21,16 @@ def self.client(options = {})
...
+      client.send(method, *args, **keywargs, &block)
+    end
+  else
+    def self.method_missing(method, *args, &block)
+      return super unless client.respond_to?(method)
```
Comment (NARKOZ) [inline lib/gitlab.rb:31]: ditto

**Review by NARKOZ — CHANGES_REQUESTED** [review]
Thank you for the PR! I'll merge it after the comments get addressed.

File: lib/gitlab.rb (hunk 1 above)
Comment (atog) [reply lib/gitlab.rb:26]: 👍🏻

File: lib/gitlab.rb (hunk 2 above)
Comment (atog) [reply lib/gitlab.rb:32]: 👍🏻

File: lib/gitlab.rb [hunk 3]
```diff
@@ -21,11 +21,18 @@ def self.client(options = {})
     Gitlab::Client.new(options)
   end
 
-  # Delegate to Gitlab::Client
-  def self.method_missing(method, *args, &block)
-    return super unless client.respond_to?(method)
-
-    client.send(method, *args, &block)
+  if RUBY_VERSION >= '3.0.0'
```
Comment (asedge) [inline lib/gitlab.rb:24]: I think you should use `Gem::Version` to compare the version strings. See doc [here](https://docs.ruby-lang.org/en/2.6.0/Gem/Version.html).

Comment (atog) [reply lib/gitlab.rb]: You are right, that approach is a much better. Thanks.

**Review by NARKOZ — APPROVED** [review]

**Comment by NARKOZ** [comment]
Thank you ❤️

**Pull request merged**